tests/Browser/PilotPassportChallengeTest.php

```diff
@@ -0,0 +1,145 @@
+            $browser->visit('/pilot_passport')
+                    ->clickLink('Settings')
+                    ->assertSee('ZTL understands that some pilots may want to participate');
+        });
+    }
+
+    public function test_change_privacy_settings(): void {
+        $this->browse(function (Browser $browser) {
+            $browser->visit('/pilot_passport')
+                    ->clickLink('Settings')
+                    ->radio('@privacy', '2')
+                    ->press('Save Settings')
+                    ->assertSee('Your privacy preferences have been saved.');
+        });
+    }
+
+    public function test_disenroll_purge(): void {
+        $this->browse(function (Browser $browser) {
+            $browser->visit('/pilot_passport')
+                    ->clickLink('Settings')
+                    ->press('@purge_data')
+                    ->whenAvailable('.modal', function (Browser $modal) {
+                        $modal->type('@confirm', 'confirm - purge all')
+                            ->press('Continue');
+                    })
+                    ->waitFor('.navbar', 30)
+                    ->assertSee('You have been successfully logged out');
+        });
+    }
+
+    public function loginSamplePilot(): void {
+        $this->checkSamplePilot();
+        $this->browse(function (Browser $browser) {
+            $browser->loginAs(Config::get('vatsim.auth_dev_credential'), 'realops');
+        });
+    }
+
+    public function checkSamplePilot(): void {
+        $u = RealopsPilot::find(Config::get('vatsim.auth_dev_credential'));
+        if (!$u) {
+            $u = new RealopsPilot;
+            $u->id = Config::get('vatsim.auth_dev_credential');
+            $u->fname = 'Web';
+            $u->lname = 'Two';
+            $u->email = 'wm@ztlartcc.org';
+            $u->save();
+        }
+        $this->clearSamplePilotActivity($u->id);
+        $this->addSamplePilotAccomplishments($u->id);
+    }
+
+    public function clearSamplePilotActivity($cid): void {
+        PilotPassportEnrollment::where('cid', $cid)->delete();
+        PilotPassportLog::where('cid', $cid)->delete();
+        PilotPassportAward::where('cid', $cid)->delete();
+    }
+
+    public function addSamplePilotAccomplishments($cid): void {
+        if (!PilotPassportLog::where('cid', $cid)->exists()) {
+            $l = new PilotPassportLog;
+            $l->id = 1;
+            $l->cid = $cid;
+            $l->airfield = self::TEST_AIRPORT_ID;
+            $l->visited_on = date('Y-m-d H:i:s');
+            $l->callsign = 'RPA3573';
+            $l->aircraft_type = 'E75L';
+            $l->save();
+        }
+        if (!PilotPassportAward::where('cid', $cid)->exists()) {
+            $a = new PilotPassportAward;
+            $a->id = 1;
+            $a->cid = $cid;
+            $a->challenge_id = self::TEST_CHALLENGE_ID;
+            $a->awarded_on = date('Y-m-d H:i:s');
+            $a->save();
+        }
+    }
+}
```

Comment:
These are good tests to start and test the frontend, but I think there are some gaps here that would be better suited for unit testing on the backend

Reply:
I'm interested to hear your thoughts on this